src/main/java/org/ibp/api/rest/study/GermplasmStudySourceResource.java

```diff
@@ -0,0 +1,86 @@
+package org.ibp.api.rest.study;
+
+import com.google.common.base.Preconditions;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import org.apache.commons.lang3.StringUtils;
+import org.generationcp.middleware.pojos.SortedPageRequest;
+import org.generationcp.middleware.service.api.study.germplasm.source.GermplasmStudySourceDto;
+import org.generationcp.middleware.service.api.study.germplasm.source.GermplasmStudySourceSearchRequest;
+import org.ibp.api.domain.common.PagedResult;
+import org.ibp.api.domain.study.GermplasmStudySourceTable;
+import org.ibp.api.java.impl.middleware.common.validator.BaseValidator;
+import org.ibp.api.java.study.GermplasmStudySourceService;
+import org.ibp.api.rest.common.PaginatedSearch;
+import org.ibp.api.rest.common.SearchSpec;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+import javax.annotation.Resource;
+import java.util.List;
+
+@Api(value = "Study Germplasm Harvest Services")
+@Controller
+@RequestMapping("/crops")
+public class GermplasmStudySourceResource {
+
+	@Resource
+	private GermplasmStudySourceService germplasmStudySourceService;
+
+	@ApiOperation(value = "It will retrieve all generated germplasm in a study",
+		notes = "It will retrieve all generated germplasm in a study")
+	@PreAuthorize("hasAnyAuthority('ADMIN','BREEDING_ACTIVITIES','MANAGE_STUDIES')")
+	@RequestMapping(value = "/{cropname}/programs/{programUUID}/studies/{studyId}/germplasm-sources/table", method = RequestMethod.POST)
+	@ResponseBody
+	public ResponseEntity<GermplasmStudySourceTable> getGermplasmStudySourceTable(final @PathVariable String cropname,
+		@PathVariable final String programUUID,
+		@PathVariable final Integer studyId, @RequestBody final GermplasmStudySourceSearchRequest germplasmStudySourceSearchRequest) {
+
+		BaseValidator.checkNotNull(germplasmStudySourceSearchRequest, "params cannot be null");
+		final SortedPageRequest sortedRequest = germplasmStudySourceSearchRequest.getSortedRequest();
+		BaseValidator.checkNotNull(sortedRequest, "sortedRequest inside params cannot be null");
+		final String sortOrder = germplasmStudySourceSearchRequest.getSortedRequest().getSortOrder();
+		final boolean isSortOrderValid = "ASC".equalsIgnoreCase(sortOrder) || "DESC".equalsIgnoreCase(sortOrder) || StringUtils.isEmpty(sortOrder);
```

Comment:
Ideally, we want a common validator that validates value for sortOrder but https://ibplatform.atlassian.net/browse/IBP-3899 will be addressed on 15.4